include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_reduce_then_scan.h

```diff
@@ -756,12 +756,17 @@ __parallel_transform_reduce_then_scan(oneapi::dpl::__internal::__device_backend_
                                       _ScanInputTransform __scan_input_transform, _WriteOp __write_op, _InitType __init,
                                       _Inclusive, _IsUniquePattern)
 {
-    using _CustomName = oneapi::dpl::__internal::__policy_kernel_name<_ExecutionPolicy>;
-    using _ReduceKernel = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<
-        __reduce_then_scan_reduce_kernel<_CustomName>>;
-    using _ScanKernel = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_provider<
-        __reduce_then_scan_scan_kernel<_CustomName>>;
     using _ValueType = typename _InitType::__value_type;
+    using _CustomName = oneapi::dpl::__internal::__policy_kernel_name<_ExecutionPolicy>;
+    using _ReduceKernel = oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<
```

Comment:
A nitpick. I am curious if constants such as `__sub_group_size` and `__max_inputs_per_item` should also be included into the kernel names. I think they are not strictly necessary right now since `__sub_group_size` is the same, and `__max_inputs_per_item` depends on the input type, which is already indirectly included into the name. I would suggest leaving a comment about it...

Reply:
You are correct that these are not necessary right now. I did not include them, so we do not further increase the size of the kernel name which is user unfriendly (especially when profiling). However, if this ever changes, then we need to be careful here, so I have added a comment.